src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -679,6 +716,56 @@ private static async Task<bool> IsAuthorityAliasedAsync(RequestContext requestCo
 
                 return result.Aliases.Any(alias => alias.Equals(requestAuthorityInfo.Host));
             }
+
+            private static AuthorityInfo GetMtlsAdjustedAuthorityInfo(RequestContext requestContext, AuthorityInfo configAuthorityInfo)
+            {
+                bool useMtlsPop = requestContext != null && requestContext.UseMtlsPop;
+                string authorityPath = configAuthorityInfo.CanonicalAuthority.AbsolutePath;
+                string authorityHost = configAuthorityInfo.Host;
+
+                // Check if the authority tenant is "common", meaning no specific authority was provided by the user
+                if (useMtlsPop && authorityPath.Contains("/common", StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new MsalClientException(
+                        MsalError.AuthorityHostMismatch,
+                        MsalErrorMessage.MtlsCommonAuthorityNotAllowedMessage);
+                }
+
+                // Select the appropriate mTLS endpoint based on the cloud environment
+                if (useMtlsPop)
+                {
+                    string mtlsHost;
+                    if (authorityHost.Equals("login.microsoftonline.com", StringComparison.OrdinalIgnoreCase))
```

Comment:
Should you define constants for these strings?
"login.microsoftonline.com"
"mtlsauth.microsoft.com"
"login.microsoftonline.us"

etc. Surely these are used other places in the .NET codebase?

Reply:
+1. These is not a good list of authorities. [RegionAndMtlsDiscoveryProvider.cs](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4965/files#diff-c735b57f62d49a5181a154bc10c53f62d9d5f8cba92afb7c99eaa8a3f88953c2) already has similar logic. Just reuse that.